draft-ietf-quic-transport.md

```diff
@@ -2858,8 +2858,8 @@ discussed in more detail in {{QUIC-RECOVERY}}.
 
 To limit ACK Blocks to those that have not yet been received by the sender, the
 receiver SHOULD track which ACK frames have been acknowledged by its peer.  Once
-an ACK frame has been acknowledged, the packets it acknowledges SHOULD NOT be
-acknowledged again.
+an ACK frame has been acknowledged, an endpoint SHOULD omit all ACK Blocks that
+have not changed since the ACK frame was sent from future ACK frames.
```

Comment:
Cant this be reworded somehow? I think I know what it means, but I don't understand the text.

Reply:
Certainly, suggestions welcome.

Reply:
"The receiver SHOULD track which ACK frames have been acknowledged by its
peer in order to exclude already acknowledged packets from future ACK frames
whenever these packets would unnecessarily contribute to the ACK frame size."

I mention packets and not ranges or blocks because sometimes joining two blocks can reduce the size.

I wonder if adding "have been acknowledged or presumed lost" would make any sense.

Reply:
I'm happy with your suggested text.  It's less proscriptive, but fairly clear in intent.  @kazuho does that work for you?

Reply:
@ianswett It does.

Reply:
Updated, while keeping a bit of the existing text, PTAL.

Reply:
My rewording was to avoid a double SHOULD, but this is also fine.

Reply:
Perhaps a clarification is needed that packets can be repeated until peers sees. For example by adding ", but can/MAY otherwise repeat packets for redundancy.".

To limit ACK Blocks to those that have not yet been received by the sender,
the receiver SHOULD track which ACK frames have been acknowledged.  The
receiver SHOULD exclude already acknowledged packets from future ACK frames
whenever these packets would unnecessarily contribute to the ACK frame size, but can otherwise repeat packets for redundancy."